feat: Add audio parameter support to gemini tts models

AyrennC opened the pull request:
## Title

Add 'audio' params support to all gemini tts models

## Relevant issues

Fixes #11250 
Fixes #11118 

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

![image](https://github.com/user-attachments/assets/415ebdcf-48be-4b3f-9e40-fe1c0c798995)

## Type

🆕 New Feature
🐛 Bug Fix
✅ Test

## Changes

- Add is_model_gemini_audio_model() method to detect TTS models
- Include 'audio' parameter in supported params for TTS models
- Map OpenAI audio parameter to Gemini speechConfig format
- Add assistant message transformation for Gemini audio output

AyrennC committed:
feat: Add Gemini TTS audio parameter support

- Add is_model_gemini_audio_model() method to detect TTS models
- Include 'audio' parameter in supported params for TTS models
- Map OpenAI audio parameter to Gemini speechConfig format
- Add _extract_audio_response_from_parts() method to transform audio
  output to openai format

AyrennC commented:
All tests in [LiteLLM Mock Tests Run](https://github.com/BerriAI/litellm/actions/runs/15362420847/job/43231206348?pr=11287) succeeded up to 99% before timing out at 8 minutes. 

There were no failures or code issues.

![image](https://github.com/user-attachments/assets/22581a49-cddf-49b4-bcba-be80e82b902a)

AyrennC commented:
- Squashed commits for a cleaner history

- Tested gemini tts models locally to be working, gemini currently only support `pcm16` audio format:

![image](https://github.com/user-attachments/assets/5a424bcc-0afa-4d54-8253-19434728fca3)

AyrennC committed:
updated unit-test to use pcm16

AyrennC commented:
- Squashed commits for a cleaner history
- Tested gemini tts models locally to be working, gemini currently only support `pcm16` audio format:
  ![image](https://github.com/user-attachments/assets/cce7de1b-8243-4c09-a970-a9a954f630e5)

AyrennC commented:
[LiteLLM Mock Tests](https://github.com/BerriAI/litellm/actions/runs/15364292113/job/43235199601?pr=11287) timed out after 8 minutes, all test were successful.

krrishdholakia commented on tests/llm_translation/test_gemini_tts.py:
please move this test inside `test_litellm/`

AyrennC replied:
moved to `test_litellm/`

krrishdholakia commented on litellm/llms/gemini/chat/transformation.py:
```diff
@@ -67,6 +67,20 @@ def __init__(
+        # Check if model contains "tts" and is a Gemini model
+        if "tts" in model:
+            # Check if it's in the gemini_models list
+            if model in litellm.gemini_models:
```
could just be `if "tts" in model: return True`

AyrennC replied:
updated to `if "tts" in model: return True`

krrishdholakia commented on litellm/llms/gemini/chat/transformation.py:
```diff
@@ -97,6 +113,40 @@ def map_openai_params(
+                        )
+
+                    # Map OpenAI audio parameter to Gemini speech config
+                    speech_config = {}
```
can we have these be typed dict's inside `types/llms/vertex_ai.py` - so any future updates are also tracked correctly

AyrennC replied:
added typed dict for SpeechConfig and its child in `types/llms/vertex_ai.py`

krrishdholakia reviewed (changes requested):
migrate test to test_litellm, and simplify tts model check 

rest looks great. thank you for your work on this

AyrennC committed:
- created typedict for speechconfig
- simplified gemini tts model detection
- moved gemini_tts test to test_litellm

AyrennC commented:
> migrate test to test_litellm, and simplify tts model check
> 
> rest looks great. thank you for your work on this

Thanks! I went ahead and made the suggested changes:
- created typedict for speechconfig
- simplified tts model detection
- moved test to test_litellm
  ![image](https://github.com/user-attachments/assets/36faedef-4bc8-4116-9680-5905c3e7c8fe)

AyrennC committed:
simplified is_model_gemini_audio_model more

krrishdholakia commented:
Thanks @AyrennC would you mind contributing docs for the change, so people know how to use this? 

For VertexAI - [here](https://github.com/BerriAI/litellm/blob/13dc757873141ad1def210a2046282c9ddea912a/docs/my-website/docs/providers/vertex.md#L4)
For Google AI Studio - [here](https://github.com/BerriAI/litellm/blob/13dc757873141ad1def210a2046282c9ddea912a/docs/my-website/docs/providers/gemini.md#L4)

Contributing guide - https://docs.litellm.ai/docs/extras/contributing (although it's just an .md change, so i'm sure you can just do it on github as well)
